test/e2e/openliberty_sso.go

```diff
@@ -0,0 +1,364 @@
+
+	t.Logf("Namespace: %s", namespace)
+
+	f := framework.Global
+
+	// create one replica of the operator deployment in current namespace with provided name
+	err = e2eutil.WaitForOperatorDeployment(t, f.KubeClient, namespace, "open-liberty-operator", 1, retryInterval, operatorTimeout)
+	if err != nil {
+		util.FailureCleanup(t, f, namespace, err)
+	}
+
+	if err = testSocialLogin(t, f, ctx); err != nil {
+		util.FailureCleanup(t, f, namespace, err)
+	}
+
+	if err = testProviderLogins(t, f, ctx); err != nil {
+		util.FailureCleanup(t, f, namespace, err)
+	}
+
+}
+
+// Test simple social login and various runtime behaviour
+func testSocialLogin(t *testing.T, f *framework.Framework, ctx *framework.TestCtx) error {
+	ns, err := ctx.GetNamespace()
+	if err != nil {
+		return err
+	}
+
+	const name string = "openliberty-sso"
+
+	// Create Secret for Github Login
+	secretTarget := types.NamespacedName{Name: name+"-olapp-sso", Namespace: ns}
+	data := map[string][]byte{
+		"github-clientId": []byte("bW9vb29vb28="),
+		"github-clientSecret": []byte("dGhlbGF1Z2hpbmdjb3c="),
+		"oidc-clientId": []byte("bW9vb29vb28="),
+		"oidc-clientSecret": []byte("dGhlbGF1Z2hpbmdjb3c="),
+	}
+
+	err = util.CreateSecretForSSO(f, ctx, secretTarget, data)
+	if err != nil {
+		return err
+	}
+	// Set up certificates for route, see: cert-manager test in RC
+	// terminationPolicy := v1.TLSTerminationReencrypt
+	expose := true
+	clusterIp := corev1.ServiceTypeClusterIP
+	githubLogin := v1beta1.GithubLogin{Hostname: "github.com"}
+	openliberty := util.MakeBasicOpenLibertyApplication(t, f, name, ns, 1)
+	openliberty.Spec.ApplicationImage = "openliberty/open-liberty:full-java8-openj9-ubi"
+	// TODO debug why these environment variables are causing it to crash loop
+	openliberty.Spec.Env = []corev1.EnvVar{
+		{Name: "SEC_TLS_TRUSTDEFAULTCERTS", Value: "true"},
+		{Name: "SEC_IMPORTS_K8S_CERTS", Value: "true"},
+	}
+	openliberty.Spec.Service = &v1beta1.OpenLibertyApplicationService{
+		Type: &clusterIp,
+		Port: 9080,
+		Certificate: &v1beta1.Certificate{},
+	}
+	openliberty.Spec.Expose = &expose
+	openliberty.Spec.SSO = &v1beta1.OpenLibertyApplicationSSO{
+		Github: &githubLogin,
+	}
+	err = f.Client.Create(goctx.TODO(), openliberty, &framework.CleanupOptions{TestContext: ctx, RetryInterval: time.Second, Timeout: time.Second })
+	if err != nil {
+		return err
+	}
+
+	err = e2eutil.WaitForDeployment(t, f.KubeClient, ns, name, 1, retryInterval, timeout)
+	if err != nil {
+		return err
+	}
+
+	t.Log("verifying environment variables of containers")
+
+	target := types.NamespacedName{Name: name, Namespace: ns}
+	if err = verifyEnvVariables(ctx, f, target, secretTarget); err != nil {
+		return err
+	}
+	t.Log("secret data applied successfully")
+
+	// TODO: Update secret and verify values
+	secret := corev1.Secret{}
+	err = f.Client.Get(goctx.TODO(), secretTarget, &secret)
+	if err != nil {
+		return err
+	}
+
+	secret.Data = map[string][]byte{
+		"github-clientId": []byte("different="),
```

Comment:
Please use base64 encoded values

Reply:
After some experimenting, this [post](https://github.com/kubernetes/client-go/issues/403#issuecomment-386760053) seems accurate in that the JSON encoding done by the go client encodes these as base64. So the test is more accurate with just the original strings when creating the Secret through the go client.